### PR TITLE
fixed bugs in relval_step.py and added VBF Hbb Hinv 2018 workflows

### PR DESCRIPTION
**Pull request by chayanit**
Dear experts, 

I have added VBFHToZZTo4Nu_M125 and VBFHToBB_M125 2018 workflows as they are missing and needed for HCAL changes study (according to https://its.cern.ch/jira/browse/PDMVRELVALS-4). I have also found some bugs in relval_steps.py where geometry is missing in Digi, Reco, Harvest steps and PU Minbias was not added in RecoPU17 step  for 2017 relvals. 

Please consider and merge these changes for PdmV. Thank you. 

Cheers,
Chayanit as L3 validation manager in PdmV

### CODE REVIEW

**Comment by prebello** [comment]
Hi @chayanit , did you contact someone else to know about the 'Geom' : 'DB:Extended' in 2017?
in 2017 you have to check the file
Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py
and you can see that for 2017 option there is 
upgradeProperties[2017] = {
    '2017' : {
        'Geom' : 'DB:Extended',
        'GT' : 'auto:phase1_2017_realistic',
        'HLTmenu': '@relval2017',
	'Era' : 'Run2_2017',
       'ScenToRun' : ['GenSimFull','DigiFull','RecoFull','HARVESTFull','ALCAFull'], 

threrefore this new inclusion of 'Geom' : 'DB:Extended', is not needed in principle, unless you have added wfs in a different way. 
Please take a look in your new wfs before testing this PR.

**Comment by zhenhu** [comment]
Hi Patricia, the workflows Chayanit updated in this PR are used to produce the special LHE relvals. They are the standard workflows rather than the upgrade ones. I’ll double check to make sure this PR works properly.

**Comment by prebello** [comment]
Hi @zhenhu I know and I also know that new 2017 relvals are added as "standard" instead of "upgrade" because it seems to be easier, and the implementation looks not wrong.
What I am pointing here is:
1) is it actually a bug in 2017 conditions? 
2) how this change in Geometry affect other 2017 wfs? was it checked?
We know from past experience that changing code here affects things elsewhere.. so just to be careful when saying that "a bug was fixed". ;-)  
FYI @chayanit

**Comment by prebello** [comment]
Other thing @zhenhu  is baseDataSetRelease[13] string for the input right, acc our latest PR with updates?

**Comment by prebello** [comment]
@zhenhu  @chayanit  your injections based on this PR is wrong, 
CMSSW_10_4_0_patch1__fullsimPU_2018_JMEforHCALchanges_VBF-1550206669
CMSSW_10_4_0_patch1__fullsimPU_2017_JMEforHCALchanges_VBF-1550206307
acc
https://cmsweb.cern.ch/couchdb/reqmgr_config_cache/fd52f8388d9216eb70af7c83a4d0a084/configFile
it is using
--pileup_input das:/RelValMinBias_13/CMSSW_10_4_0-103X_mc2017_realistic_v2-v1/GEN-SIM 
please close it and reject your batches.
Please, before injecting, make a test to check cmsdrivers. After all ends ok, then you can make PR for implementing changes. 
Thank you.

**Comment by zhenhu** [comment]
Hi @prebello , yes, baseDataSetRelease[13] in this PR should be updated to 'CMSSW_10_5_0_pre1-103X_mc2017_realistic_v2-v1' to be consistent with the upgrade2017 workflows. 
But the previous injections by Chayanit seem fine to me, since they were submitted in 104X, and the strings used there were also for 104X.

**Comment by chayanit** [comment]
Hi @prebello, I updated string at baseDataSetRelease[13] to the one I used for 10_4_0_patch1 campaign (CMSSW_10_4_0-103X_mc2017_realistic_v2-v1). I'm not sure why we need to do this as the string at baseDataSetRelease[13] is not used anymore for our main campaign (10_5_X). Anyway let me know if everything is ok now.

**Comment by kpedro88** [comment]
please test

**Comment by prebello** [comment]
> Hi @prebello, I updated string at baseDataSetRelease[13] to the one I used for 10_4_0_patch1 campaign (CMSSW_10_4_0-103X_mc2017_realistic_v2-v1). I'm not sure why we need to do this as the string at baseDataSetRelease[13] is not used anymore for our main campaign (10_5_X). Anyway let me know if everything is ok now.

Hi @chayanit, you have to update with the new 10-5-0-pre1 strings instead (the latest regenerated), not 10-4-0-patch1(used before). The reason: you added new wfs that will recycle GEN-SIM  for PU. Making this PR means that your updates will come in next releases and these relvals have to read theis proper inputs (not from 10-4-X but from 10-5-X). Is it clear?

**Review by prebello — CHANGES_REQUESTED** [review]
please use 'CMSSW_10_5_0_pre1-103X_mc2017_realistic_v2-v1' to be consistent with the upgrade2017 workflows.

**Comment by zhenhu** [comment]
please test workflow 1362.17,1362.18,25212.17,25212.18

**Comment by zhenhu** [comment]
+1

File: Configuration/PyReleaseValidation/python/relval_steps.py [hunk 1]
```diff
@@ -628,6 +628,8 @@ def identitySim(wf):
     'CMSSW_10_5_0_pre1-103X_upgrade2018_realistic_v8-v1',  # 18 - GENSIM input for 2018 fullSim premix workflows
     'CMSSW_10_5_0_pre1-103X_upgrade2018_realistic_v8_FastSim-v1',    # 19 - fastSim MinBias for mixing UP18
     'CMSSW_10_5_0_pre1-PU25ns_103X_upgrade2018_realistic_v8_FastSim-v1',# 20 - fastSim premix library UP18 
+    'CMSSW_10_4_0-103X_mc2017_realistic_v2-v1',		# 21 - GEN-SIM inputs for LHE-GEN-SIM 2017 workflows
```
Comment (prebello) [inline Configuration/PyReleaseValidation/python/relval_steps.py]: perfect. thank you

File: Configuration/PyReleaseValidation/python/relval_steps.py [hunk 2]
```diff
@@ -628,6 +628,8 @@ def identitySim(wf):
...
+    'CMSSW_10_4_0-103X_upgrade2018_realistic_v8-v1',		# 22 - GEN-SIM inputs for LHE-GEN-SIM 2018 workflows
```
Comment (prebello) [inline Configuration/PyReleaseValidation/python/relval_steps.py]: perfect too

**Comment by fabiocos** [comment]
@kpedro88 this update is not upgrade specific, could you please anyway have a look? It seems ok to me

**Comment by fabiocos** [comment]
+1

**Comment by kpedro88** [comment]
+upgrade

**Pull request merged**